### PR TITLE
avoid not specialized Pair issue

### LINKED ISSUE #2888: mixing `:x => :y` and `:x => f => :y` syntax in vector to `select` errors
MWE:

```julia
using DataFrames

tab = DataFrame(x = 1:10, y = 1:10)

xselector = :x => ByRow(x -> 2x) => :x2
yselector = :y => :Y
y2selector = :y => ByRow(y -> 2y) => :y2

# all okay
select(tab, xselector)
select(tab, yselector)
select(tab, xselector, yselector)
select(tab, (xselector, yselector)...)

# error:
select(tab, [xselector, yselector])
select(tab, [xselector, yselector]...)

# okay:
select(tab, [xselector, y2selector])
select(tab, [xselector, y2selector]...)
```

Errors are like

```
ArgumentError: Unrecognized column selector: Pair{Symbol, Any}(:x, ByRow{var"#3#4"}(var"#3#4"()) => :x2)

Stacktrace:
 [1] normalize_selection(idx::DataFrames.Index, sel::Any, renamecols::Bool)
   @ DataFrames ~/.julia/packages/DataFrames/vuMM8/src/abstractdataframe/selection.jl:184
 [2] manipulate(::DataFrame, ::Any, ::Vararg{Any, N} where N; copycols::Bool, keeprows::Bool, renamecols::Bool)
   @ DataFrames ~/.julia/packages/DataFrames/vuMM8/src/abstractdataframe/selection.jl:1311
 [3] #manipulate#470
   @ ~/.julia/packages/DataFrames/vuMM8/src/abstractdataframe/selection.jl:1438 [inlined]
 [4] select(df::DataFrame, args::Any; copycols::Bool, renamecols::Bool)
   @ DataFrames ~/.julia/packages/DataFrames/vuMM8/src/abstractdataframe/selection.jl:940
 [5] select(df::DataFrame, args::Any)
   @ DataFrames ~/.julia/packages/DataFrames/vuMM8/src/abstractdataframe/selection.jl:940
 [6] top-level scope
   @ In[21]:1
 [7] eval
   @ ./boot.jl:360 [inlined]
 [8] include_string(mapexpr::typeof(REPL.softscope), mod::Module, code::String, filename::String)
   @ Base ./loading.jl:1094
```

### PR DESCRIPTION
**Pull request by bkamins**
Fixes #2888

### CODE REVIEW

File: src/abstractdataframe/selection.jl [hunk 1]
```diff
@@ -182,6 +182,14 @@ end
 # add a method to funname defined in other/utils.jl
 funname(row::ByRow) = funname(row.fun)
 
+function make_pair_concrete(@nospecialize(x))
+    if x isa Pair
+        return make_pair_concrete(x.first) => make_pair_concrete(x.second)
+    else
+        return x
+    end
+end
```
Comment (nalimilan) [inline src/abstractdataframe/selection.jl:191]: Isn't this equivalent and more idiomatic?
```suggestion
make_pair_concrete(@nospecialize(x::Pair)) =
    make_pair_concrete(x.first) => make_pair_concrete(x.second)
make_pair_concrete(@nospecialize(x)) = x
```

File: test/select.jl [hunk 2]
```diff
@@ -1678,4 +1678,18 @@ end
     @test select(df, :a => ByRow(f) => :a) == DataFrame(a=1:3)
 end
 
+@testset "making Pair concrete" begin
```
Comment (nalimilan) [inline test/select.jl:1681]:
```suggestion
@testset "vectors of pairs with non-concrete type are accepted" begin
```

**Review by nalimilan — APPROVED** [review]

File: src/abstractdataframe/selection.jl (hunk 1 above)
Comment (bkamins) [reply src/abstractdataframe/selection.jl:191]: I was not sure if it works this way, but it seems it does :).

**Comment by bkamins** [comment]
Thank you!

**Pull request merged**